index.js

```diff
@@ -139,19 +139,19 @@ ShopifyToken.prototype.getAccessToken = function getAccessToken(shop, code, fn)
     var status = response.statusCode
       , body = '';
 
-    if (status !== 200) {
-      return fn(new Error('Invalid status code (' + status + ') returned'));
-    }
-
     response.setEncoding('utf8');
     response.on('data', function data(chunk) {
       body += chunk;
     });
     response.on('end', function end() {
+      if (status !== 200) {
+        return fn(new Error('Failed to get Shopify access token, status: ' + status + ', ' + body));
```

Comment:
Instead of putting all on the error message I would add two properties to the error object.

- `error.responseBody` to store the response body.
- `error.statusCode` to store the response status code.